3-8 Branch(write excel module is good except the lunar calendar

feiger1992 opened the pull request:
the format of codes after today will be same

feiger1992 committed:
Merge pull request #2 from feiger1992/new-branch

the input of extreme values

feiger1992 committed:
tide read

I delete it heedless

feiger1992 committed:
for debug

feiger1992 committed:
change some module

feiger1992 committed:
something about the NaN values

the new way to calculate the avarage of timedelta
the module don't have direct way to do it

feiger1992 committed:
Delete tide read.py

feiger1992 committed:
filter again

filter the hight and lowest values

feiger1992 committed:
lunar calendar

lunar calendar
move the position of higgest/lowest values

duboviy commented on Qt35.py:
```diff
@@ -0,0 +1,318 @@
+                                                                                            (
+                                                                                                temp_data.if_min == True) & (
+                                                                                                temp_data.format_time < x)].format_time.max(), 'tide']
+                except:
```
- To catch any exception use the `except Exception`, instead of `except BaseException` or just `except`.

duboviy commented on Qt35.py:
```diff
@@ -0,0 +1,318 @@
+                        (temp_data.if_max == True) & (temp_data.format_time < x)].format_time.max()
+                    temp_data.loc[x, 'diff'] = temp_data.loc[temp_data[(temp_data.if_max == True) & (
+                        temp_data.format_time < x)].format_time.max(), 'tide'] - temp_data.loc[x, 'tide']
+                except:
```
- To catch any exception use the `except Exception`, instead of `except BaseException` or just `except`.

duboviy commented on Qt35.py:
```diff
@@ -0,0 +1,318 @@
+                        pass
+                    else:
+                        say_out('Please Check Your Format of the Sheet')
+            except:
```
- To catch any exception use the `except Exception`, instead of `except BaseException` or just `except`.

feiger1992 replied:
ok
thx

duboviy reviewed (changes requested):
- To catch any exception use the `except Exception`, instead of `except BaseException` or just `except`.
- Indicate the most specific exception type in except block:
```python
     # Bad
     try:
         mapping[key]
     except Exception:
         ...

     # Better
     try:
         mapping[key]
     except KeyError:
         ...
```

feiger1992 committed:
change some format
add the sheet of init_data

feiger1992 committed:
repair some bug:
1. get  highest/lowest tide in the series
2. only one_month_long data process
     :right_button conor is not perfect

feiger1992 committed:
add new module about stream data process

feiger1992 committed:
add new module about stream data process
a lot TODO

feiger1992 committed:
增加两种输出格式

feiger1992 committed:
生成报表

feiger1992 committed:
生成报表，补充第一个高/低潮位

feiger1992 committed:
界面

feiger1992 committed:
界面更新

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
处理气象数据（DAT）

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
raising/ebbing time calculation process repair

feiger1992 committed:
潮流模块全部重写

feiger1992 committed:
界面增加了一部分

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
各层矢量输出

feiger1992 committed:
增加潮流模块，图片输出

feiger1992 committed:
加入文字显示特征值等其他功能

feiger1992 committed:
加入其他显示功能

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
改进了部分单元格的输入问题
1.每天的平均值以及总计
2.每月对应的统计值，之前为默认round

feiger1992 committed:
打开文件选择的时候改成了记忆最近打开的位置

feiger1992 committed:
改正分层过程

feiger1992 committed:
增加沉底式流速数据处理以及两层时采用幂函数推算表层的计算方法

feiger1992 committed:
增加了沉底式测流的选项

feiger1992 committed:
预处理函数可以选择处理站点

feiger1992 committed:
Add files via upload

增加功能

feiger1992 committed:
new modules of read onboard data

new functions to plot in current

feiger1992 committed:
Update README.md
